tests/thread_local/enc/enc.cpp

```diff
@@ -75,27 +75,59 @@ class spinlock
 
 // Assert that each thread-local variable has a unique address across threads.
 std::set<volatile void*> g_addresses;
-spinlock g_addresses_lock;
+spinlock g_lock;
+
+// Total number of threads.
+volatile int g_total_num_threads = 0;
+
+// Current number of threads.
+volatile int g_num_threads = 0;
 
 void assert_unique_address(volatile void* var_address)
 {
-    g_addresses_lock.acquire();
+    g_lock.acquire();
     OE_TEST(g_addresses.count(var_address) == 0);
     g_addresses.insert(var_address);
-    g_addresses_lock.release();
+    g_lock.release();
 }
 
 // Clear test data
-void clear_test_data()
+void prepare_for_test(int total_num_threads)
 {
-    g_addresses_lock.acquire();
+    g_lock.acquire();
     g_addresses.clear();
-    g_addresses_lock.release();
+    g_total_num_threads = total_num_threads;
+    g_num_threads = 0;
+    g_lock.release();
+}
+
+void increment_num_threads()
+{
+    g_lock.acquire();
+    OE_TEST(++g_num_threads <= g_total_num_threads);
+    g_lock.release();
+}
+
+void wait_for_test_completion()
+{
+    // Wait for all threads to complete
+    bool complete = false;
+    while (!complete)
+    {
+        // Sleep for sometime.
+        host_usleep(10);
+
+        g_lock.acquire();
+        complete = (g_num_threads == g_total_num_threads);
```

Comment:
Oh cool so this is the final condition for the threads to terminate, once 16 of them have been created. Makes sense and looks good to me!

Reply:
Yes.